README.md

```diff
@@ -276,3 +276,79 @@ mav_trajectory_generation::drawMavTrajectoryWithMavMarker(trajectory, distance,
 mav_trajectory_generation::drawMavSampledTrajectoryWithMavMarker(states, distance, frame_id, hex, &markers)
 ```
 
+## Checking Feasibility
+The package contains three implementations to check generated trajectories for
+input feasibility. The checks are based on the rigid-body model assumption and
+flat state characteristics presented in
+
+```
+@inproceedings{mellinger2011minimum,
+  title={Minimum snap trajectory generation and control for quadrotors},
+  author={Mellinger, Daniel and Kumar, Vijay},
+  booktitle={Robotics and Automation (ICRA), 2011 IEEE International Conference on},
+  pages={2520--2525},
+  year={2011},
+  organization={IEEE}
+}
+```
+
+The trajectories are checked for low and high thrust, high velocities, high roll
+and pitch rates, high yaw rates and high yaw angular accelerations.
+
+`FeasibilitySampling` implements a naive sampling-based check.
+`FeasibilityRecursive` implements a slightly adapted recursive feasibility test
+presented in
+```
+@article{mueller2015computationally,
+  title={A computationally efficient motion primitive for quadrocopter trajectory generation},
+  author={Mueller, Mark W and Hehn, Markus and D'Andrea, Raffaello},
+  journal={IEEE Transactions on Robotics},
+  volume={31},
+  number={6},
+  pages={1294--1310},
+  year={2015},
+  publisher={IEEE}
+}
+```
+We adapted [https://github.com/markwmuller/RapidQuadrocopterTrajectories] to
+check arbitrary polynomial order trajectories for yaw and velocity feasibility.
+
+`FeasibilityAnalytic` analytically checks the magnitudes except for the roll and
+pitch rates, where it runs the recursive test (recommended: low in computation
+time, no false positives).
+
+###Example
+```c++
+// Create input constraints.
+InputConstraints input_constraints;
+input_constraints.setFMin(0.5); // minimum thrust in [m/s/s].
```

Comment:
m/s/s isn't thrust though; `F = m a` and then this is the max total acceleration, so a scale factor off from thrust.